webapp/src/main/java/com/box/l10n/mojito/monitoring/QuartzPendingJobsReportingTask.java

```diff
@@ -21,28 +25,49 @@
 
     private JdbcTemplate jdbcTemplate;
     private MeterRegistry meterRegistry;
+    private Map<String, AtomicLong> queueSizes;
 
     public QuartzPendingJobsReportingTask(@Autowired DataSource dataSource,
                                           @Autowired MeterRegistry meterRegistry) {
         this.meterRegistry = meterRegistry;
         this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.queueSizes = new ConcurrentHashMap<>();
     }
 
     @Scheduled(fixedRateString = "${l10n.management.metrics.quartz.sql-queue-monitoring.execution-rate}")
     public void reportPendingJobs() {
-        fetchResults().forEach(this::reportResults);
+        List<PendingJob> results = fetchResults();
+        updateQueueSizes(results);
+        results.forEach(this::registerJobQueueSize);
+    }
+
+    private void registerJobQueueSize(PendingJob pendingJob) {
+        queueSizes.computeIfAbsent(pendingJob.key, key -> createGauge(pendingJob))
+                .set(pendingJob.count);
     }
 
-    private void reportResults(PendingJob pendingJob){
-        meterRegistry.gauge("quartz.pending.jobs",
-                            Tags.of("jobClass", pendingJob.jobClass, "jobGroup", pendingJob.jobGroup),
-                            pendingJob.count);
+    private AtomicLong createGauge(PendingJob pendingJob) {
+        return meterRegistry.gauge("quartz.pending.jobs",
+                Tags.of("jobClass", pendingJob.jobClass, "jobGroup", pendingJob.jobGroup),
+                new AtomicLong(pendingJob.count));
     }
 
-    List<PendingJob> fetchResults(){
+    private void updateQueueSizes(List<PendingJob> pendingJobs) {
+        queueSizes.forEach((key, val) -> {
+            Long size = pendingJobs.stream()
+                    .filter(pj -> pj.key.equals(key))
```

Comment:
use a Map instead of a list for `List<PendingJob>` to have constant search time instead of iterating on the list?